### PR TITLE
AN-6765: App bypasses Do Not Disturb settings

### PR DESCRIPTION
**Pull request by BradleyWire**
## What's new in this PR?

JIRA: https://wearezeta.atlassian.net/browse/AN-6765

### Issues

When do not disturb is turned on we're still able to receive interruptions from calls through Wire. 

### Causes

The tone plays when the intensity of the volume is max which doesn't take into account the do not disturb settings. 

### Solutions

Listen out of changes in the do not disturb settings and play the corresponding call tone + the notification based on the priority. 

### Testing

Scenario 1 (Do not disturb is on + accept call interruptions from no one)
1. Go to do not disturb settings + toggle on 
2. Sign in to User A on any device 
3. Sign in to User B on Android device 
4. Leave app in background 
5. User A should ring User B
6. No ringtone should be triggered but a banner notification will still appear (depending on manufacturer) 

Scenario 2 (Do not disturb is on + accept call interruptions from everyone)
1.  Go to do not disturb settings + toggle on 
2. Only accept interruptions from everyone who is calling 
3. Sign in to User A on any device 
4. Sign in to User B on Android device 
5. Leave app in background 
6. User A should ring User B
7. Ringtone should be triggered and banner notification will appear from the top down and in the banner notification box

Scenario 3 (Do not disturb is on + accept call interruptions from Wire) 
1. Go to do not disturb settings + toggle on 
2. Go to apps 
3. Go to wire 
4. Go to notifications 
5. Bypass Do not disturb setting for call notifications should be on 
6. Sign in to User A on any device 
7. Sign in to User B on Android device 
8. Leave app in background 
9. User A should ring User B
10. Ringtone should be triggered and banner notification will appear from the top down and in the banner notification box
#### APK
[Download build #2101](http://10.10.124.11:8080/job/Pull%20Request%20Builder/2101/artifact/build/artifact/wire-dev-PR2841-2101.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/common/controllers/SoundController.scala [hunk 1]
```diff
@@ -185,7 +203,7 @@ class SoundControllerImpl(implicit inj: Injector, cxt: Context)
   }
 
   def playRingFromThemInCall(play: Boolean): Unit =
-    setMediaPlaying(R.raw.ringing_from_them_incall, play)
+    setMediaPlaying(if (shouldPlayCallTone) R.raw.ringing_from_them_incall else 0, play)
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/common/controllers/SoundController.scala:206]: Can this `0` here be instead a `val` defined in the object?

Comment (BradleyWire) [reply app/src/main/scala/com/waz/zclient/common/controllers/SoundController.scala]: I can move the if outside of this parameter anyway by the looks of it without having the 0

**Comment by BradleyWire** [comment]
build again

File: app/src/main/scala/com/waz/zclient/common/controllers/SoundController.scala [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
+  * Wire
+  * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (mejdoo) [inline app/src/main/scala/com/waz/zclient/common/controllers/SoundController.scala:3]: maybe change this to 2020 :)

File: app/src/main/scala/com/waz/zclient/common/controllers/SoundController.scala [hunk 3]
```diff
@@ -95,6 +96,22 @@ class SoundControllerImpl(implicit inj: Injector, cxt: Context)
 
   def currentTonePrefs: (String, String, String) = tonePrefs.currentValue.getOrElse((null, null, null))
 
+  private def shouldPlayCallTone = notificationManager.getCurrentInterruptionFilter match {
```
Comment (mejdoo) [inline app/src/main/scala/com/waz/zclient/common/controllers/SoundController.scala:99]: well done 👍

**Pull request merged**